File: src/main/java/com/example/demo/Item.java
package com.example.demo;

public class Item {

    private String itemName;
    private double itemCost;
    private int quantity;

    public Item(String itemName, double itemCost, int quantity) {
        this.itemName = itemName;
        this.itemCost = itemCost;
        this.quantity = quantity;
    }

    public String getItemName() {
        return this.itemName;
    }
    public double getItemCost(){
        return this.itemCost;
    }
    public int getQuantity(){
        return this.quantity;
    }
}

File: frontend/src/app/models/item.ts
export class Item {
    private itemName: string;
    private itemCost: number;
    private quantity: number;
    private itemTotal: number;

    constructor(
        itemName: string,
        itemCost: number,
        quantity: number,
        itemTotal: number
    ) {
        this.itemName = itemName;
        this.itemCost = itemCost;
        this.quantity = quantity;
        this.itemTotal = this.quantity * this.itemCost;
    }

    public getItemName = (): string => {
        return this.itemName;
    }

    public getItemCost = (): number => {
        return this.itemCost;
    }

    public getQuantity = (): number => {
        return this.quantity;
    }

    public getItemTotal = (): number => {
        return this.itemTotal;
    }

    public getData = (): Item => {
        return this;
    }

}
